src/az_ulib_ipc/az_ulib_ipc.c

```diff
@@ -0,0 +1,352 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
+
+#include <stdint.h>
+
+#include "azure_macro_utils/macro_utils.h"
+#include "umock_c/umock_c_prod.h"
+
+#include "az_pal_os_api.h"
+#include "az_ulib_action_api.h"
+#include "az_ulib_descriptor_api.h"
+#include "az_ulib_ipc_api.h"
+#include "az_ulib_base.h"
+#include "internal/az_ulib_ipc.h"
+#include "ucontract.h"
+#include "ulib_config.h"
+#include "ulib_port.h"
+#include "ulib_result.h"
+#include "ulog.h"
+
+/*
+ * IPC is a singleton component, and shall be initialized only once.
+ */
+static _az_ulib_ipc *ipc = NULL;
+
+static _az_ulib_ipc_interface *get_interface_by_name_and_version(
```

Comment:
>get_interface_by_name_and_version [](start = 31, length = 33)

nit: get_interface ? since both name and version are parameters, I don't think we need to describe the parameters here in the name.